lib/tjbot.js

```diff
@@ -51,14 +50,7 @@ function TJBot(hardware, configuration, credentials) {
     }
 
     // import configuration params
-    this.configuration = {};
-    TJBot.prototype.configurationParameters.forEach(function(param) {
-        if (configuration != undefined && configuration[param] != undefined) {
-            this.configuration[param] = configuration[param];
-        } else {
-            this.configuration[param] = TJBot.prototype.defaultConfiguration[param];
-        }
-    }, this);
+    this.config = configuration;
```

Comment:
I am in support of full names as a general principle, but I think config works here (also inline with the naming of config.js).